server/cli/src/cli.rs

```diff
@@ -502,29 +507,81 @@ async fn main() -> anyhow::Result<()> {
                         (None, None) => None,
                     };
 
-                    if let Some(form_schema_json) = &form_schema_json {
-                        FormSchemaRowRepository::new(&con).upsert_one(form_schema_json)?;
-                    }
-
-                    ReportRowRepository::new(&con).upsert_one(&ReportRow {
+                    let report_data = ReportData {
                         id,
                         name: report_name,
                         r#type: repository::ReportType::OmSupply,
-                        template: fs::read_to_string(report_path)?,
+                        template: report_definition,
                         context,
                         sub_context,
-                        argument_schema_id: form_schema_json.map(|r| r.id.clone()),
+                        argument_schema_id: form_schema_json.clone().map(|r| r.id.clone()),
                         comment: None,
                         is_custom,
                         version: version.to_string(),
                         code: report_code,
-                    })?;
+                        form_schema: form_schema_json,
+                    };
+
+                    reports_data.reports.push(report_data);
                 }
             }
 
+            let output_path = match code.clone() {
+                Some(code) => format!("{base_reports_dir}/generated/{code}.json"),
+                None => format!("{base_reports_dir}/generated/standard_reports.json"),
+            };
+            let output_path = Path::new(&output_path);
+
+            fs::create_dir_all(output_path.parent().ok_or(anyhow::Error::msg(format!(
+                "Invalid output path: {:?}",
+                output_path
+            )))?)?;
+
+            fs::write(output_path, serde_json::to_string_pretty(&reports_data)?).map_err(|_| {
+                anyhow::Error::msg(format!(
+                    "Failed to write to {:?}. Does output dir exist?",
+                    output_path
+                ))
+            })?;
+
             match code.clone() {
-                Some(code) => info!("{}", format!("{code} report upserted")),
-                None => info!("All standard reports upserted"),
+                Some(code) => info!("{}", format!("{code} report built")),
+                None => info!("All standard reports built"),
+            }
+        }
+        Action::UpsertReportsJson { json_path } => {
+            let connection_manager = get_storage_connection_manager(&settings.database);
+            let con = connection_manager.connection()?;
+            let base_reports_dir = "./reports";
+            let generated_dir = format!("{base_reports_dir}/generated");
+
+            let json_file = fs::File::open(
+                json_path.unwrap_or(format!("{generated_dir}/standard_reports.json")),
+            )
+            .expect("json not found");
```

Comment:
```suggestion
            .expect("{generated_dir}/standard_reports.json not found");
```
Would be nice to be more explicit with error case...